pyaedt/modules/PostProcessor.py

```diff
@@ -601,7 +601,351 @@ def surfacePlotInstruction(self):
+
+    """
+
+    def __init__(self, parent):
+        self._parent = parent
+        self._scratch = Scratch(self._parent.temp_directory, volatile=True)
+
+    @property
+    def oreportsetup(self):
+        """Report setup.
+
+        Returns
+        -------
+        :attr:`pyaedt.modules.PostProcessor.PostProcessor.oreportsetup`
+
+        """
+        return self.odesign.GetModule("ReportSetup")
+
+    @property
+    def _messenger(self):
+        """Messenger."""
+        return self._parent._messenger
+
+    @property
+    def _desktop(self):
+        """Desktop."""
+        return self._parent._desktop
+
+    @property
+    def odesign(self):
+        """Design."""
+        return self._parent._odesign
+
+    @property
+    def oproject(self):
+        """Project."""
+        return self._parent._oproject
+
+    @property
+    def modeler(self):
+        """Modeler."""
+        return self._parent._modeler
+
+    @property
+    def oeditor(self):
+        """Editor."""
+        return self.modeler.oeditor
+
+
+    @property
+    def post_solution_type(self):
+        """Design solution type.
+
+        Return
+        ------
+        type
+            Design solution type.
+        """
+        try:
+            return self.odesign.GetSolutionType()
+        except:
+            return self._parent._design_type
+
+    @aedt_exception_handler
+    def copy_report_data(self, PlotName):
+        """Copy report data as static data.
+
+        Parameters
+        ----------
+        PlotName :
+            Name of the report.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.CopyReportsData([PlotName])
+        self.oreportsetup.PasteReports()
+        return True
+
+    @aedt_exception_handler
+    def delete_report(self, PlotName):
+        """Delete a field plot report.
+
+        Parameters
+        ----------
+        PlotName : str
+            Name of the field plot report.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.DeleteReports([PlotName])
+        return True
+
+    @aedt_exception_handler
+    def rename_report(self, PlotName, newname):
+        """Rename a plot.
+
+        Parameters
+        ----------
+        PlotName : str
+            Name of the plot.
+        newname : str
+            New name of the plot.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.RenameReport(PlotName, newname)
+        return True
+
+    @aedt_exception_handler
+    def get_report_data(self, expression="dB(S(1,1))", setup_sweep_name='', domain='Sweep', families_dict=None,
+                        report_input_type=None):
+        """Generate report data using the :func:`GetSolutionDataPerVariation` method.
```

Comment:
This method can't be linked by sphinx, so it would be best to remove this entirely.

```suggestion
        """Generate report data.
```